src/main/java/com/wolox/training/models/User.java

```diff
@@ -0,0 +1,94 @@
+package com.wolox.training.models;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+import com.wolox.training.exceptions.BookAlreadyOwnedException;
+import com.wolox.training.exceptions.NotFoundException;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Pattern;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+@Entity(name = "Users")
+public class User {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private long id;
+
+    @NotNull
+    private String username;
+
+    @NotNull
+    private String name;
+
+    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern="yyyy-MM-dd")
+    @NotNull
+    private LocalDate birthdate;
+
+    @ManyToMany(cascade = CascadeType.ALL)
+    private List<Book> books = new ArrayList<>();
+
+    public User(){
+
+    }
+
+    public User(@NotNull String username, @NotNull String name, @NotNull LocalDate birthdate) {
+        this.username = username;
+        this.name = name;
+        this.birthdate = birthdate;
+    }
+
+    public long getId() {
+        return id;
+    }
+
+    public String getUsername() {
+        return username;
+    }
+
+    public void setUsername(String username) {
+        this.username = username;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public LocalDate getBirthdate() {
+        return birthdate;
+    }
+
+    public void setBirthdate(LocalDate birthdate) {
+        this.birthdate = birthdate;
+    }
+
+    public List<Book> getBooks() {
+        return Collections.unmodifiableList(this.books);
+    }
+
+    public void setBooks(List<Book> books) {
+        this.books = books;
+    }
+
+    public void addBook(Book book) throws BookAlreadyOwnedException {
+        if(this.books.contains(book)){
+            throw new BookAlreadyOwnedException("The User already owns the Book");
+        }
+        this.books.add(book);
+    }
+
+    public void removeBook(Book book) {
+        if(!(this.books.contains(book))){
```

Comment:
Just an observation, the parenthesis are not necessary after the `!` symbol: 
`if (!books.contains(book)) {` will suffice 😄  
Adding `this` is not necessary either in this case :)